src/core/request.js

```diff
@@ -44,13 +44,15 @@ Request.prototype.request = function(servers, callback_) {
   const callback = typeof servers === 'function' ? servers : callback_;
 
   this.emit('before');
+
+  const wasRequested = this.requested;
```

Comment:
Seems like unnecessary complexity unless there is a good use case, in which case we should add a comment to explain.